src/matrix.rs

```diff
@@ -51,6 +51,45 @@ macro_rules! matrix {
         impl<T> AsRef<[T; $inner * $outer]> for $name<T> {
             fn as_ref(&self) -> &[T; $inner * $outer] { unsafe { ::core::mem::transmute(self) } }
         }
+
+        #[cfg(feature = "serde")]
+        impl<T> ::serde::Serialize for $name<T>
+            where T: ::serde::Serialize
+        {
+            fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+                where S: ::serde::Serializer
+            {
+                AsRef::<[[T; $inner]; $outer]>::as_ref(self).serialize(serializer)
+            }
+        }
+
+        #[cfg(feature = "serde")]
+        impl<'de, T> ::serde::Deserialize<'de> for $name<T>
+            where T: ::serde::Deserialize<'de>
+        {
+            fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
```

Comment:
why don't we piggy-back on fixed-size slices here as well? something like:
```rust
[[T; $inner]; $outer].deserialize(D).map($name<T>::from)
```

Reply:
I thought about that, I'm just not sure if the extra conversion is worth it ? Maybe Rust will optimize it to a no-op.

Reply:
We can check if it does optimize this by looking at the assembly, even through play.rust-lang.org